### ISSUE TITLE
`routine` shouldn't fail the check status if it's out of date

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/shiny-workflows#30]
I think it's these lines that throw an error if the routine workflow runs concurrently – say if you push a few commits one after the other. 
It adds a lot of noise to workflow runs, to the point I find I'm just dismissing these notifications.

https://github.com/rstudio/shiny-workflows/blob/7dd795b564c2463c544cc5d2bcb83e166e23fd33/.github/internal/verify-no-new-commits/action.yaml#L17-L30

I think this should be a warning but the workflow should ultimately succeed. I'm not sure if we can pick a different check run status, like `stale`, but that would also be nice.

**Comment by schloerke** [comment]
I believe we should solve this with concurrency setting in GHA.

What about adding something to the routine job:

```
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
  cancel-in-progress: true
```

(docs from https://docs.github.com/en/actions/using-jobs/using-concurrency)

**Comment by gadenbuie** [comment]
Yeah that's another good approach but it still results in a ❌ 
https://github.com/orgs/community/discussions/8336

**Comment by gadenbuie** [comment]
Oh man, I just realized the failure actually comes from these lines

https://github.com/rstudio/shiny-workflows/blob/286813b2fa65ffbf8eb6a2be15dab0603e2b0d92/.github/workflows/routine.yaml#L182-L186

**Comment by schloerke** [comment]
If we keep as is... too many failures, that's bad. But the checks are there for a reason, so they should make noise.
If we use concurrency... it's reported as a failure. So no better than the current solution.

If I see green lights from prior runs (who wanted to push back results, but were ignored as more commits were made) and just make a news change, I'll merge the PR before the latest run has finished.
This will cause the merged-commit to fail as there will be new changes created from the merged-commit routine run.

I can't guarantee that the routine workflow has passed in every repo unless we manually do that and feels like it would cause more headache.

I don't know how to make the Routine workflow produce less errors while maintaining full accuracy of the checks being performed.

------------------

If the discussion ever returns a [`skip-in-progress` concurrency field](https://github.com/orgs/community/discussions/8336#discussioncomment-3579579), I believe it would satisfy both of our 
requirements.

**Comment by gadenbuie** [comment]
Yeah there really isn't a good answer and I guess of all the current choices it's probably better to live with the X. It's just super annoying.
